atomix/cluster/src/test/java/io/atomix/raft/storage/log/RaftLogTest.java

```diff
@@ -0,0 +1,166 @@
+    reader = raftlog.openReader();
+    data.order(ByteOrder.LITTLE_ENDIAN).putInt(123456);
+  }
+
+  @AfterEach
+  void tearDown() {
+    reader.close();
+    raftlog.close();
+  }
+
+  @Test
+  void shouldAppendInitialEntry() {
+    // given
+    final RaftLogEntry entry = new RaftLogEntry(1, initialEntry);
+    // when
+    final var appended = raftlog.append(entry);
+
+    // then
+    assertThat(reader.hasNext()).isTrue();
+    final var entryRead = reader.next();
+    assertThat(entryRead.term()).isEqualTo(1);
+    assertThat(entryRead.entry()).isInstanceOf(InitialEntry.class);
+    assertThat(appended).isEqualTo(entryRead);
+  }
+
+  @Test
+  void shouldAppendConfigurationEntry() {
+    // given
+    final RaftLogEntry entry = new RaftLogEntry(1, configurationEntry);
+    // when
+    final var appended = raftlog.append(entry);
+
+    // then
+    assertThat(reader.hasNext()).isTrue();
+    final var entryRead = reader.next();
+    assertThat(entryRead.term()).isEqualTo(1);
+    assertThat(entryRead.entry()).isInstanceOf(ConfigurationEntry.class);
+  }
+
+  @Test
+  void shouldAppendApplicationEntry() {
+    // given
+    final RaftLogEntry entry = new RaftLogEntry(1, applicationEntry);
+    // when
+    final var appended = raftlog.append(entry);
+
+    // then
+    assertThat(reader.hasNext()).isTrue();
+    final var entryRead = reader.next();
+    assertThat(entryRead.term()).isEqualTo(1);
+    assertThat(entryRead.isApplicationEntry()).isTrue();
+    assertThat(entryRead.getApplicationEntry().lowestPosition()).isEqualTo(1);
+    assertThat(entryRead.getApplicationEntry().highestPosition()).isEqualTo(2);
+    assertThat(entryRead.getApplicationEntry().data().getInt(0)).isEqualTo(data.getInt(0));
+    assertThat(appended).isEqualTo(entryRead);
+  }
+
+  @Test
+  void shouldDeleteAfter() {
+    // given
+    raftlog.append(new RaftLogEntry(1, applicationEntry));
+    final var secondEntry = raftlog.append(new RaftLogEntry(1, applicationEntry));
+    raftlog.append(new RaftLogEntry(1, applicationEntry));
+
+    // when
+    raftlog.deleteAfter(secondEntry.index());
+
+    // then
+    assertThat(raftlog.getLastIndex()).isEqualTo(secondEntry.index());
+    assertThat(raftlog.getLastEntry()).isEqualTo(secondEntry);
+  }
+
+  @Test
+  void shouldNotDeleteCommittedEntries() {
+    // given
+    raftlog.append(new RaftLogEntry(1, applicationEntry));
+    final var deleteIndex = raftlog.append(new RaftLogEntry(1, applicationEntry)).index();
+    final var commitIndex = raftlog.append(new RaftLogEntry(1, applicationEntry)).index();
+
+    // when
+    raftlog.setCommitIndex(commitIndex);
+
+    // then
+    assertThatThrownBy(() -> raftlog.deleteAfter(deleteIndex))
+        .isInstanceOf(IllegalStateException.class);
+  }
+
+  @Test
+  void shouldReset() {
+    // given
+    raftlog.append(new RaftLogEntry(1, applicationEntry));
+    final var secondEntry = raftlog.append(new RaftLogEntry(1, applicationEntry));
```

Comment:
```suggestion
    raftlog.append(new RaftLogEntry(1, applicationEntry));
```

The variable is unused